presto-native-execution/presto_cpp/main/PrestoTask.cpp

```diff
@@ -795,20 +795,26 @@ void PrestoTask::updateExecutionInfoLocked(
   }
   for (int i = 0; i < veloxTaskStats.pipelineStats.size(); ++i) {
     auto& veloxPipeline = veloxTaskStats.pipelineStats[i];
-    if (veloxPipeline.inputPipeline) {
-      const auto& firstVeloxOpStats = veloxPipeline.operatorStats[0];
-      prestoTaskStats.rawInputPositions += firstVeloxOpStats.rawInputPositions;
-      prestoTaskStats.rawInputDataSizeInBytes +=
-          firstVeloxOpStats.rawInputBytes;
-      prestoTaskStats.processedInputPositions +=
-          firstVeloxOpStats.inputPositions;
-      prestoTaskStats.processedInputDataSizeInBytes +=
-          firstVeloxOpStats.inputBytes;
-    }
-    if (veloxPipeline.outputPipeline) {
-      const auto& lastVeloxOpStats = veloxPipeline.operatorStats.back();
-      prestoTaskStats.outputPositions += lastVeloxOpStats.outputPositions;
-      prestoTaskStats.outputDataSizeInBytes += lastVeloxOpStats.outputBytes;
+
+    // tasks may fail before any operators are created;
+    // collect stats only when we have operators
+    if (!veloxPipeline.operatorStats.empty()) {
+      if (veloxPipeline.inputPipeline) {
+        const auto& firstVeloxOpStats = veloxPipeline.operatorStats[0];
+        prestoTaskStats.rawInputPositions +=
+            firstVeloxOpStats.rawInputPositions;
+        prestoTaskStats.rawInputDataSizeInBytes +=
+            firstVeloxOpStats.rawInputBytes;
+        prestoTaskStats.processedInputPositions +=
+            firstVeloxOpStats.inputPositions;
+        prestoTaskStats.processedInputDataSizeInBytes +=
+            firstVeloxOpStats.inputBytes;
+      }
+      if (veloxPipeline.outputPipeline) {
+        const auto& lastVeloxOpStats = veloxPipeline.operatorStats.back();
+        prestoTaskStats.outputPositions += lastVeloxOpStats.outputPositions;
+        prestoTaskStats.outputDataSizeInBytes += lastVeloxOpStats.outputBytes;
+      }
```

Comment:
Suggestion: we can include two following for loops under this condition as well up to line 833